x86/boards.txt

```diff
@@ -1,7 +1,87 @@
-virtual.name="Kaleidoscope Virtual Keyboard"
-virtual.build.usb_product="Kaleidoscope Virtual Keyboard"
-virtual.build.usb_manufacturer="Kaleidoscope"
-virtual.build.board=VIRTUAL
-virtual.build.core=virtual
-virtual.build.variant=virtual
-virtual.build.extra_flags=-DKALEIDOSCOPE_HARDWARE_H="Kaleidoscope-Hardware-Virtual.h"
+# See: http://code.google.com/p/arduino/wiki/Platforms
+
+#menu.cpu=Processor
+#menu.bootloader=Bootloader
+#menu.pinout=Pinout
+
+##############################################################
+
+model01.name=Keyboardio Model 01
+model01.build.usb_product="Model 01"
+model01.build.usb_manufacturer="Keyboardio"
+model01.build.board=AVR_MODEL01
+model01.build.core=virtual
+model01.build.variant=model01
+model01.build.extra_flags={build.usb_flags} '-DKALEIDOSCOPE_HARDWARE_SPEC_H="kaleidoscope/hardware/keyboardio/Model01_Spec.h"' '-DKALEIDOSCOPE_MCU_H="mcu/atmega32u4.h"'
```

Comment:
For this approach to work, hardware headers are divided up into two headers where on defines everything that is independent from avr details, such as #rows, #cols #LEDs, LED index mappings, keymap definition macros. This stuff will further on live in a scpecification header, e.g. `Model01_Spec.h`. The rest of the original hardware header is left in place for the non-virtual builds. The specification header is included at the top of the traditional hardware header to prevent redundant definitions.

Reply:
This PR also adds virtual EEPROM. Its size is read from an additional MCU-header that will be shared by all boards of the same type. Other properties of a MCU that might be relevant to virtual builds might be added in the future.